src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,172 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
+    public MyLinkedList() {
+    }
+
     @Override
     public void add(T value) {
+        linkLastElement(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexToAdd(index);
+        if (index == size) {
+            linkLastElement(value);
+        } else {
+            linkBeforeElement(value, searchForNode(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list.isEmpty()) {
+            return;
+        }
+        Node<T> previous = last;
+        T[] elements = (T[]) list.toArray();
```

Comment:
The use of unchecked type conversion `(T[]) list.toArray()` can lead to a `ClassCastException` at runtime if the list contains elements that are not of type `T`. Consider using `list.toArray(new T[0])` if `T` is reifiable, or handle the conversion more safely.